jenkinsci/datadog-plugin
# Support environment variables in global tags and global job tags

### What does this PR do?



Add support for environment variables in global job tags as per [Issue 54](https://github.com/jenkinsci/datadog-plugin/issues/54). Support for environment variables in the global tags field was also added.  The assumption is that env vars used in these fields refer to env vars configured on the `master` jenkins agent. 

### Description of the Change



There are two cases. 
1. Freestyle jobs
2. Pipeline jobs

- For both of these cases, we can use `EnvVars.masterEnvVars` to access the environment variables
- Both of these cases are tested via unit tests and manual testing. 
- Pipeline jobs may be ran on an agent, so this was tested manually and verified for correctness. 
### Alternate Designs



### Possible Drawbacks



This only supports env variables that were defined on the master jenkins agent. But this seems to be fine and more practical than using the env vars on the agent(s). 

### Verification Process


1. Configure env variables and global job tags on master accordingly. 
2. Configure a freestyle/pipeline project. 
3. Build the project
4. Observe that global job tags are applied correctly. 

### Additional Notes



### Release Notes



### Review checklist (to be filled by reviewers)

- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
